src/legacy/shrink.rs

```diff
@@ -0,0 +1,396 @@
+use std::collections::VecDeque;
+use std::io::{self, Error, Read};
+
+use bitstream_io::{BitRead, BitReader, Endianness, LittleEndian};
+
+const MIN_CODE_SIZE: u8 = 9;
+const MAX_CODE_SIZE: u8 = 13;
+
+const MAX_CODE: usize = (1 << MAX_CODE_SIZE) - 1;
+const CONTROL_CODE: usize = 256;
+const INC_CODE_SIZE: u16 = 1;
+const PARTIAL_CLEAR: u16 = 2;
+
+// const HASH_BITS: usize = MAX_CODE_SIZE + 1; /* For a load factor of 0.5. */
+// const HASHTAB_SIZE: usize = 1 << HASH_BITS;
+const UNKNOWN_LEN: u16 = u16::MAX;
+
+struct CodeQueue {
+    next_idx: usize,
+    codes: Vec<Option<u16>>,
```

Comment:
Again, array or boxed slice. In this case it'll also simplify the code because you won't have to unwrap an Option<Option<_>>.

Reply:
fixed.